test/models/order_test.rb

```diff
@@ -106,5 +106,85 @@
         expect(order.cart_checkout).must_be_nil
       end
     end
+
+    describe "class methods " do
+      let(:merchant) { merchants(:merchant_2) }
+      let(:item_hash) { Order.find_merchant_order_items(merchant) }
+      describe "self.find_merchant_order_items(merchant, items_hash: {\"paid\" => {}, \"complete\" => {} })" do
+        it "will return a nested hash structure thatsstores merchant order items:
+        key is order status and value is a hash with order_id as key and array of item objects as value:" do
+          expect(item_hash.to_s).must_equal ({ "paid" => { "615724322" => [Item.find(477087474)] }, "complete" => { "1035625630" => [Item.find(845260767), Item.find(728299111)] } }).to_s
+        end
+
+        it "will return empty hash for values with no orders in an order.status" do
+          expect(Order.find_merchant_order_items(merchants(:merchant_3)).to_s).must_equal ({ "paid" => {}, "complete" => {} }).to_s
+        end
+
+        it "will return hash with additional statuses and add orders to them passed as params" do
+          expect(Order.find_merchant_order_items(merchant, items_hash: { "paid" => {}, "complete" => {}, "pending" => {} }).to_s).must_equal ({ "paid" => { "615724322" => [Item.find(477087474)] }, "complete" => { "1035625630" => [Item.find(845260767), Item.find(728299111)] }, "pending" => { "330565047" => [Item.find(34296661)] } }).to_s
+        end
+
+        # NOT 100% SURE IF THESE TESTS ARE ACTUALLY TESTING WHAT THEY ARE SUPPOSED TO.
+        # ALSO LOTS OF CHECKS AND VAILDATIONS, SO THESE SCENARIOS WOULD NOT HAPPEN. DO # THESE EVEN HAVE TO BE TESTED?
+
+        # it "invalid order will not be added to hash" do
+        #   order = Order.new(status: "noneya")
+        #   item = Item.new(product: products(:product_2), order: order, quantity: 4)
+        #   expect(item.valid?).must_equal true
+        #   expect(order.valid?).must_equal false
+        #   expect(item_hash.to_s).must_equal ({ "paid" => { "615724322" => [Item.find(477087474)] }, "complete" => { "1035625630" => [Item.find(845260767), Item.find(728299111)] } }).to_s
+        # end
+
+        # it "invalid item will not be added to hash" do
+        #   items(:item_2).update(quantity: nil)
+        #   expect(items(:item_2).valid?).must_equal false
+        #   expect(item_hash.to_s).must_equal ({ "paid" => { "615724322" => [Item.find(477087474)] }, "complete" => { "1035625630" => [Item.find(845260767)] } }).to_s
+        # end
```

Comment:
I agree that this test (line 138) is not testing what it says it will. The one above it (starting on line 130) seems to be but it is duplicated and refined below. I think it's safe to delete these two.